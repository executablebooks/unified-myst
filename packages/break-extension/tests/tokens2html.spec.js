import { micromark } from 'micromark'

import { mystBreakMmarkExt, mystBreakHtmlExt } from '../src/index.js'

const fixtures = [
    '+',
    '+++',
    '+++\n',
    '+++\na',
    '+++{}',
    '+++{}\n',
    '+++ {"key": "value"}',
]

describe('Convert source to HTML', () => {
    test.each(fixtures)(`%j`, (content) => {
        const options = {
            extensions: [mystBreakMmarkExt],
            htmlExtensions: [mystBreakHtmlExt],
        }
        const htmlText = micromark(content, options)
        expect(htmlText).toMatchSnapshot()
    })
})
