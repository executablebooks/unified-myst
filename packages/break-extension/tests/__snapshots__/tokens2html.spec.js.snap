// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Convert source to HTML "+" 1`] = `
"<ul>
<li></li>
</ul>"
`;

exports[`Convert source to HTML "+++ {\\"key\\": \\"value\\"}" 1`] = `"<hr class=\\"myst-break\\"/>"`;

exports[`Convert source to HTML "+++" 1`] = `"<hr class=\\"myst-break\\"/>"`;

exports[`Convert source to HTML "+++\\n" 1`] = `
"<hr class=\\"myst-break\\"/>
"
`;

exports[`Convert source to HTML "+++\\na" 1`] = `
"<hr class=\\"myst-break\\"/>
<p>a</p>"
`;

exports[`Convert source to HTML "+++{}" 1`] = `"<hr class=\\"myst-break\\"/>"`;

exports[`Convert source to HTML "+++{}\\n" 1`] = `
"<hr class=\\"myst-break\\"/>
"
`;
